projects/cm_mcu/LocalTasks.c

```diff
@@ -30,6 +31,452 @@
+    // internal statuses on page 0 : table 4.5
+    {1, 0x00, 0x0C, 1, "SYSINCAL", 0x01, "", SM_STATUS},      // page 0x00
+    {1, 0x00, 0x0C, 1, "LOSXAXB", 0x02, "", SM_STATUS},       // page 0x00
+    {1, 0x00, 0x0C, 1, "LOSREF", 0x04, "", SM_STATUS},        // page 0x00
+    {1, 0x00, 0x0C, 1, "LOS", 0x08, "", SM_STATUS},           // page 0x00
+    {1, 0x00, 0x0C, 1, "SMBUS_TIMEOUT", 0x20, "", SM_STATUS}, // page 0x00
+    {1, 0x00, 0x0D, 1, "LOSIN_IN0", 0x01, "", SM_STATUS},     // page 0x00
+    {1, 0x00, 0x0D, 1, "LOSIN_IN1", 0x02, "", SM_STATUS},     // page 0x00
+    {1, 0x00, 0x0D, 1, "LOSIN_IN2", 0x04, "", SM_STATUS},     // page 0x00
+    {1, 0x00, 0x0D, 1, "LOSIN_FB_IN", 0x08, "", SM_STATUS},   // page 0x00
+};
+
+uint16_t clkr0a_values[NSUPPLIES_CLKR0A * NPAGES_CLKR0A * NCOMMANDS_CLKR0A];
+
+struct MonitorI2CTaskArgs_t clockr0a_args = {
+    .name = "CLKR0A",
+    .devices = clkr0a_moni2c_addrs,
+    .i2c_dev = I2C_DEVICE_CLK,
+    .n_devices = NSUPPLIES_CLKR0A,
+    .commands = sm_command_clkr0a,
+    .n_commands = NCOMMANDS_CLKR0A,
+    .n_values = NSUPPLIES_CLKR0A * NPAGES_CLKR0A * NCOMMANDS_CLKR0A,
+    .n_pages = NPAGES_CLKR0A,
+    .sm_values = clkr0a_values,
+    .smbus = &g_sMaster2,
+    .smbus_status = &eStatus2,
+    .xSem = NULL,
+    .requirePower = true,
+    .stack_size = 4096U,
+};
+
+/*
+TickType_t getFFupdateTick(){
+  return ffl12_f1_args.updateTick;
+}
+*/
+
+int8_t getFFtemp(const uint8_t i)
+{
+  int i1 = 1;
+  int8_t val;
+  configASSERT(i < NFIREFLIES);
+  if (i < NFIREFLIES_IT_F1) {
+    int index = i * (ffl12_f1_args.n_commands * ffl12_f1_args.n_pages) + i1;
+    val = ffl12_f1_args.sm_values[index];
+  }
+
+  else if (NFIREFLIES_IT_F1 <= i && i < NFIREFLIES_IT_F1 + NFIREFLIES_DAQ_F1) {
+    int index = (i - NFIREFLIES_IT_F1) * (ffldaq_f1_args.n_commands * ffldaq_f1_args.n_pages) + i1;
+    val = ffldaq_f1_args.sm_values[index];
+  }
+
+  else if (NFIREFLIES_F1 <= i && i < NFIREFLIES_F1 + NFIREFLIES_IT_F2) {
+    int index = (i - NFIREFLIES_F1) * (ffl12_f2_args.n_commands * ffl12_f2_args.n_pages) + i1;
+    val = ffl12_f2_args.sm_values[index];
+  }
+  else {
+    int index = (i - NFIREFLIES_F1 - NFIREFLIES_IT_F2) * (ffldaq_f2_args.n_commands * ffldaq_f2_args.n_pages) + i1;
+    val = ffldaq_f2_args.sm_values[index];
+  }
+
+  return val;
+}
+
+void getFFpart()
+{
+
+  if (ffldaq_f1_args.xSem != NULL) {
+    while (xSemaphoreTake(ffldaq_f1_args.xSem, (TickType_t)10) == pdFALSE)
+      ;
+  }
+  // Write device vendor part for identifying FF devices
+  // FF connecting to FPGA1
+  uint8_t vendor_data1[4];
+  uint32_t vendor_char1;
+  int8_t vendor_part1[16];
+
+  // FF connecting to FPGA2
+  uint8_t vendor_data2[4];
+  uint32_t vendor_char2;
+  int8_t vendor_part2[16];
+
+  uint8_t ven_addr_start = VENDOR_START_BIT_FF12;
+  uint8_t ven_addr_stop = VENDOR_STOP_BIT_FF12;
+
+  // checking the FF 12-ch part connected to FPGA1 (need to check from Rx devices (i.e devices[odd]))
+  uint8_t data1[1];
```

Comment:
replace array of array 1 with just uint8_t

Reply:
Done